### PR TITLE
Introduce validatePassword for SignIn and SignUp resources

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

Introduce validatePassword for SignIn and SignUp resources to be used in custom flows

### CODE REVIEW

File: packages/clerk-js/src/core/resources/SignIn.ts [hunk 1]
```diff
@@ -234,6 +235,16 @@ export class SignIn extends BaseResource implements SignInResource {
     });
   };
 
+  validatePassword: ReturnType<typeof createValidatePassword> = (password, cb) => {
+    if (SignIn.clerk.__unstable__environment?.userSettings.passwordSettings) {
+      return createValidatePassword({
+        // TODO: Remove as any
```
Comment (desiprisg) [inline packages/clerk-js/src/core/resources/SignIn.ts:241]: Is this something you wanted to do as part of this PR?

Comment (panteliselef) [reply packages/clerk-js/src/core/resources/SignIn.ts:241]: This is how this functionality becomes part of the public api, and makes the feature available to developers who use clerk.

Comment (desiprisg) [reply packages/clerk-js/src/core/resources/SignIn.ts:241]: I meant the TODO :D

**Review by raptisj — APPROVED** [review]

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**